### ISSUE TITLE
Issues with reproducing the code results of the paper

### ISSUE THREAD

**Issue by ftd0828** [expectorlin/NavCoT#13]
Hello! Your paper and associated code are truly fascinating, and we’ve been working to reproduce the results. During this process, we’ve encountered a few questions and would greatly appreciate your guidance.

In Table 1, you mention the use of the R2R augmentation dataset. However, from the provided download link, we only found the file "R2R_train_instruction_finetuning_data.json". We’d like to confirm: when reproducing the results in Table 1, is the process to fine-tune using the pre-trained large model weights and corresponding tokenizer you provided, with "R2R_train_instruction_finetuning_data.json" as the training data?

Thank you for your time and help!

**Comment by expectorlin** [comment]
Thanks very much for your interest in our work! The "R2R_train_instruction_finetuning_data.json" can be utilized to repoduce the results of "NavCoT+LLaMA 2 (ours) " in Table 1. To reproduce the NavCoT+LLaMA 2 (ours)* results, you can attempt to obtain the instruction finetuning data for a randomly chosen R2R augmentation subset according to the following steps:

1. We utilize the ChatGPT model to extract the landmark list from the navigation instruction for each trajectory-instruction pair.

2. We obtain the imagination ground-truth (GT) for each timestep of the expert trajectory, by choosing the instruction landmark which has the maximum similarity with the ground-truth action, calculated using the CLIP model. The GT for the visual information filter (VIF) and action prediction (AP) can be directly obtained by indexing the ground-truth action in the candidate actions (observations). We utilize the imagination GT, VIF GT, and AP GT to construct the CoT GT.

3. With the CoT GT, we collect the prompt-CoT GT pair for each timestep of the expert trajectory, by setting the 'feedback' parameter as 'teacher' when running the validation code for the training dataset. The prompt is constructed by organizing the task example, navigation instruction, the observational description, and the history.

4. Based on the prompt-CoT GT pairs for each expert trajectory, we construct the finetuning datasets by referring to the format of the [alpaca_gpt4_data.json](https://github.com/Instruction-Tuning-with-GPT-4/GPT-4-LLM/blob/main/data/alpaca_gpt4_data.json).

By combining and shuffling the instruction finetuning data for the training set and R2R augmentation set, the new instruction finetuning data can be used to train the LLM to reproduce the results based on the R2R augmentation set.
